src/android.ts

```diff
@@ -46,10 +46,14 @@ export const androidApi = <RunTarget extends SupportedRunTarget<'android'>>(
         objectionProcesses: [],
 
         awaitAdb: async () => {
-            const adbIsStarted = await retryCondition(
-                async () => (await execa('adb', ['get-state'], { reject: false })).exitCode === 0,
-                100
-            );
+            const adbIsStarted = await retryCondition(async () => {
+                const { exitCode, stdout } = await execa(
+                    'adb',
+                    ['wait-for-device', 'shell', 'getprop', 'dev.bootcomplete'],
```

Comment:
I haven't found any documentation on `dev.bootcomplete`. But I have found two other scripts that additionally check for `sys.boot_completed` (https://chromium-review.googlesource.com/c/chromiumos/third_party/autotest/+/350770/2/server/hosts/adb_host.py#525, https://github.com/AzimoLabs/fastlane-plugin-automated-test-emulator-run/blob/fa9e76d7b1db914622524eae2f2abba8851d3e04/lib/fastlane/plugin/automated_test_emulator_run/actions/automated_test_emulator_run_action.rb#L321). Maybe we should do that as well?

Also, the fastlane one uses that opportunity to also check for `init.svc.bootanim == stopped`, which we are currently doing [in CA](https://github.com/tweaselORG/cyanoacrylate/blob/b_initial-version/src/util.ts#L135-L149). Maybe we should move that here as well, now that we're checking for more than adb being ready?